lib/rspec/core/output_wrapper.rb

```diff
@@ -0,0 +1,26 @@
+module RSpec
+  module Core
+    # Wrapper around an output stream object.
+    class OutputWrapper
+      # @private
+      attr_accessor :output
+
+      # @param output [IO, StringIO] an object to be wrapped
+      def initialize(output)
+        @output = output
+      end
+
+      [:<<, :advise, :autoclose=, :autoclose?, :binmode, :binmode?, :bytes, :chars, :close, :close_on_exec=,
+       :close_on_exec?, :close_read, :close_write, :closed?, :codepoints, :each, :each_byte, :each_char,
+       :each_codepoint, :each_line, :eof, :eof?, :external_encoding, :fcntl, :fdatasync, :fileno, :flush,
+       :fsync, :getbyte, :getc, :gets, :inspect, :internal_encoding, :ioctl, :isatty, :lineno, :lineno=,
+       :lines, :pid, :pos, :pos=, :print, :printf, :putc, :puts, :read, :read_nonblock, :readbyte, :readchar,
+       :readline, :readlines, :readpartial, :reopen, :rewind, :seek, :set_encoding, :stat, :sync, :sync=,
+       :sysread, :sysseek, :syswrite, :tell, :to_i, :to_io, :tty?, :ungetbyte, :ungetc, :write, :write_nonblock].each do |method|
```

Comment:
Good idea. I suppose we want to decorate only IO specific methods, not Object methods. I just have tried, `IO.instance_methods - Enumerable.instance_methods - Object.instance_methods` gives same list as above except `:inspect` method. Specs are green.